public/js/p3/widget/CircularViewerContainer.js

```diff
@@ -123,6 +124,95 @@ define([
+			var sp_query = "?eq(genome_id," + this.state.genome_ids[0] + ")" + filter + "&sort(+patric_id,+property)" + "&select(" + sp_fields.join(",") + ")&limit(25000)";
+			// console.log("sp_query: ", sp_query);
+
+			var track = this.viewer.addTrack({
+				type: SectionTrack,
+				options: {
+					title: title,
+					title_tooltip: title_tooltip,
+					loadingText: "LOADING " + title.toUpperCase(),
+					loading: true,
+					trackWidth: 0.05,
+					fill: fill,
+					stroke: stroke,
+					gap: 0,
+					background: background,
+					formatPopupContent: function(item){
+						//return item.patric_id + " (" + item.feature_type + ")<br>Product: " + item.product + "<br>Location: " + item.location;
+						return DataItemFormatter(item, "feature_data", {mini: true, linkTitle: true})
+					},
+					formatDialogContent: function(item){
+						return DataItemFormatter(item, "feature_data", {linkTitle: true})
+					}
+				}
+			})
+
+			return xhr.get(PathJoin(this.apiServiceUrl, "sp_gene", sp_query), {
+				headers: {
+					accept: "application/json",
+					'X-Requested-With': null,
+					'Authorization': (window.App.authorizationToken || "")
+				},
+				handleAs: "json"
+			}).then(lang.hitch(this, function(spgenes){
+				// console.log(" spgenes:", spgenes);				
+				if (spgenes.length == 0) {
+					track.set('loading', false);
+	                Topic.publish("/Notification", {message: "No data found.", type: "error"});                                          	
+					return spgenes;
+				} else {
+					var feature_ids = [];
+					
+					for (var i=0; i < spgenes.length; i++) {
+						feature_ids.push(spgenes[i].feature_id);
+					}
+					
+					var query = "in(feature_id,(" + feature_ids.join(",") + "))&sort(+accession,+start)" + "&select(" + fields.join(",") + ")&limit(25000)";
+					// console.log(" query:",  PathJoin(this.apiServiceUrl, "genome_feature", query));								 
+					
+					xhr.post(PathJoin(this.apiServiceUrl, "genome_feature/"), {
+						data: query,
+						headers: {
+							accept: "application/solr+json",
+							'Content-Type': "application/rqlquery+x-www-form-urlencoded",
+							'X-Requested-With': null,
+							'Authorization': (window.App.authorizationToken || "")
+						},
+						handleAs: "json"					
+					}).then(lang.hitch(this, function(res){
+						// console.log(" features:", res);
+						var refseqs = res.response.docs;
+						
+						if (refseqs.length == 0) {
+							track.set('loading', false);
+							Topic.publish("/Notification", {message: "No data found.", type: "error"});                                          
+					
+							return refseqs;
+						}
+						// console.log(" refseqs:", refseqs); 										
+						refseqs = refseqs.map(function(r){
+							r.name = r.accession;
+							r.length = r.end - r.start;
```

Comment:
i think we do have na_length as a property of feature. you don't need to calculate on the fly

Reply:
Here we use nt length

Reply:
Kept as is.